pkgpanda/__init__.py

```diff
@@ -656,18 +699,24 @@ def symlink_all(src, dest):
             # Add to the active folder
             os.symlink(package.path, os.path.join(self._make_abs("active.new"), package.name))
 
-            # Add to the environment contents
-            env_contents += "# package: {0}\n".format(package.id)
-            for k, v in package.environment.items():
-                env_contents += "{0}={1}\n".format(k, v)
-            env_contents += "\n"
+            # Add to the environment and environment.export contents
 
-            # Add to the environment.export contents
+            env_contents += "# package: {0}\n".format(package.id)
             env_export_contents += "# package: {0}\n".format(package.id)
+
             for k, v in package.environment.items():
+                env_contents += "{0}={1}\n".format(k, v)
                 env_export_contents += "export {0}={1}\n".format(k, v)
+
+            env_contents += "\n"
             env_export_contents += "\n"
 
+            # execute sysctl
+            for sysctl_var, sysctl_value in package.sysctl.items():
+                if _is_root():
```

Comment:
The way this is operating is making unexpected environment-specific differences in how things are installed which make it non-obivous why something happens / doesn't happen.

The way this particular codeabse works is that people on the outside / calling specify exactly what they want. If we're running on a final install system, we should _always_ unconditionally set the systemctls, and if they fail, we should fail to install and error loudly

Inside unit / integration tests we either use monkeypatching to stub in the root calls (And make sure they are called), or add config parameters which allow toggling the individual options on and off (Admittedly those toggles need a _lot_ of refactoring).